app/models/product.rb

```diff
@@ -61,4 +64,9 @@ def price_present?
     def discount_price_present?
       discount_price.present?
     end
+
+    def update_total_products_count_of_category
+      category.total_products_count = category.products_count
```

Comment:
will this save the category?

Reply:
No, it won't save the category
Update code to save the category